controllers/dscinitialization/utils.go

```diff
@@ -18,132 +21,178 @@ import (
 	"k8s.io/client-go/util/retry"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 
 	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
 )
 
 var (
 	resourceInterval = 10 * time.Second
 	resourceTimeout  = 1 * time.Minute
 )
 
-// createOdhNamespace creates a Namespace with given name and with ODH defaults. The defaults include:
-// - Odh specific labels
-// - Pod security labels for baseline permissions
-// - ConfigMap  'odh-common-config'
-// - Network Policies 'opendatahub' that allow traffic between the ODH namespaces
-// - RoleBinding 'opendatahub'.
-func (r *DSCInitializationReconciler) createOdhNamespace(ctx context.Context, dscInit *dsciv1.DSCInitialization, name string, platform cluster.Platform) error {
+// createOperatorResource include steps:
+// - 1. validate customized application namespace || create/update application namespace
+// - 2. patch application namespaces for managed cluster
+//   - Odh specific labels for access
+//   - Pod security labels for baseline permissions//
+//
+// - 3. Patch monitoring namespace
+// - 4. Network Policies 'opendatahub' that allow traffic between the ODH namespaces
+// - 5. ConfigMap  'odh-common-config'
+// - 6. RoleBinding 'opendatahub'.
+func (r *DSCInitializationReconciler) createOperatorResource(ctx context.Context, dscInit *dsciv1.DSCInitialization, platform cluster.Platform) error {
 	log := logf.FromContext(ctx)
-	// Expected application namespace for the given name
-	desiredNamespace := &corev1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: name,
-			Labels: map[string]string{
-				labels.ODH.OwnedNamespace: "true",
-				labels.SecurityEnforce:    "baseline",
-			},
-		},
+	nsName := dscInit.Spec.ApplicationsNamespace
```

Comment:
we should do a validation here so that the DSCI namespace matches the one labelled with the `application-namespace` label